Help Center: Re-design improvements

renancarvalho opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/9226
Fixes https://github.com/Automattic/dotcom-forge/issues/9226

## Proposed Changes

* Apply the new redesign changes CMvR9P4FontAbv2b0TNMzj-fi-282_49597 on Help Center.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Match the new design.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use live link or pull and run this branch
* Make sure to add the feature flag to see the design changes, `?flags=help-center-experience`
* Start a new chat
* Compare against the designs CMvR9P4FontAbv2b0TNMzj-fi-282_49597
* Make sure to explore around, create a new chat, use Zendesk to reply to your own messages
* Check any possible regressions without the flag

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arcangelini committed:
Odie Client Refactor

heavyweight committed:
Fix message listener

heavyweight committed:
Fix lost conversationId when sending a message

heavyweight committed:
Merge branch 'trunk' into odie/refactor-messages-component

heavyweight committed:
Fix build issue

heavyweight committed:
Fix placeholder after user message

renancarvalho committed:
Fix cleanup call back function

renancarvalho committed:
Fix textarea size after sending long message

renancarvalho committed:
Merge branch 'trunk' into odie/refactor-messages-component

escapemanuele committed:
Fix Wapuu thinking

renancarvalho committed:
Add styling for introduction wappu message

renancarvalho committed:
Add new styles for user messages

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Improve chat feedback options

renancarvalho committed:
Improve dislike styles

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Add thumbs down component

renancarvalho committed:
Render thumbs down fake message

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Improve spacing

renancarvalho committed:
Apply correct styles on messages

renancarvalho committed:
Refactor name

renancarvalho committed:
Add new send message button style

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Refactoring  message component

renancarvalho committed:
Rename variable

renancarvalho committed:
Fix logic mistake

renancarvalho committed:
Add feature flag checks

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Improvements

renancarvalho committed:
Add chat with support component

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Improve related guides

escapemanuele commented:
Thanks for working on this!

| Design | Current |
|--------|--------|
| <img width="385" alt="image" src="https://github.com/user-attachments/assets/e2dadd97-23e9-4789-8e81-6a025d62c5e0"> | <img width="385" alt="image" src="https://github.com/user-attachments/assets/6331aeb4-0a7f-4939-8dd6-725c12f1973c"> |

All the message should be inside the gray "bubble"

<img width="475" alt="image" src="https://github.com/user-attachments/assets/9fa949c9-b01d-4de9-9e6c-afdf8301c9b0">

`Related guides` needs some changes (both with and without flag)

<img width="460" alt="image" src="https://github.com/user-attachments/assets/c9ef5204-4732-4a09-af80-deb72cff4f82">

Padding should be 16px instead of 12px

arcangelini commented on packages/odie-client/src/components/message/style_redesign.scss:
Would you be able to structure this in a way that is logical with maybe some sections? What you have now is great, but I have personally been stuck refactoring styles that we didn't organize on the front end and led to excess styles, unnecessary `!important`'s and general confusion. Right now you have a great understanding of what is happening so it would be helpful to arrange it in a way that in understandable for other.

renancarvalho committed:
Add chat date component

arcangelini commented on packages/odie-client/src/components/message/get-support.scss:
What do you think would be better?

1. Having a single CSS file for the entire `message` directory
2. Having a single CSS file for every component in the directory (like we have in Help Center)

I was the one who originally added this file but I am not sure what the best option would be. Based on what you think would be best we should either, split the current SCSS file you have into individual files, or combine this into the main stylesheet. Just to maintain consistency.

renancarvalho replied:
Personally, I rather have a style per component, I find easier to deal with smaller files.

If you're up to that as well, I can split it out in a follow-up PR.

arcangelini replied:
> If you're up to that as well, I can split it out in a follow-up PR.

Absolutely. Happy to review whenever you have that one as well. I feel the exact same as you.

arcangelini commented on packages/odie-client/src/components/message/index.tsx:
```diff
@@ -104,9 +104,14 @@ const ChatMessage = ( {
+	const { botName, shouldUseHelpCenterExperience } = useOdieAssistantContext();
 	const [ isFullscreen, setIsFullscreen ] = useState( false );
 	const [ isDisliked ] = useState( false );
+	if ( shouldUseHelpCenterExperience ) {
+		import( './style_redesign.scss' );
+	} else {
+		import( './style.scss' );
+	}
```
💯 this is a great call

renancarvalho committed:
Add chat date flag

renancarvalho committed:
fix temp style loading

renancarvalho committed:
fix message input padding

arcangelini commented on packages/odie-client/src/components/message/message-content.tsx:
```diff
@@ -27,50 +29,72 @@ export const MessageContent = forwardRef<
 			'odie-chatbox-message',
-			isUser ? 'odie-chatbox-message-user' : 'odie-chatbox-message-wapuu',
+			isUser && 'odie-chatbox-message-user',
+			isWapuu && 'odie-chatbox-message-wapuu',
```
Wapuu is no longer a thing. We should be using something more generic like `odie` or `bot`. May seem trivial, but the name of all the pieces on this project has made it difficult to understand and explain (odie, wapuu, help center, zendesk, smooch, sunshine, etc)

renancarvalho replied:
I've renamed to `bot`, it makes things simpler and more readable.

arcangelini commented on packages/odie-client/src/components/message/message-content.tsx:
```diff
@@ -27,50 +29,72 @@ export const MessageContent = forwardRef<
+			isUser && 'odie-chatbox-message-user',
+			isWapuu && 'odie-chatbox-message-wapuu',
+
+			shouldUseHelpCenterExperience && isUser && 'odie-chatbox-message-user',
+			shouldUseHelpCenterExperience && isHuman && 'odie-chatbox-message-business',
+			shouldUseHelpCenterExperience && isWapuu && 'odie-chatbox-message-wapuu',
```
These seem to be redundant besides the `isHuman` one.

Also, would it be easier if it were something like:

```
`odie-chatbox-message-${ message.role }`
```

This would be more easier to understand IMO. The only change you would have to make is update the CSS 'odie-chatbox-message-wapuu' to 'odie-chatbox-message-bot' to match

renancarvalho replied:
Good suggestion, I have addressed, `isHuman` still mapped to `business`, all the others are dinamyc.

renancarvalho committed:
Fix invalid date on chatDate component

arcangelini commented on packages/odie-client/src/components/message/messages-container.tsx:
```diff
@@ -29,10 +38,20 @@ export const MessagesContainer = forwardRef< HTMLDivElement, ChatMessagesProps >
 			if ( message.type === 'error' ) {
 				lastErrorMessageIndex = index;
 			}
+			// Find the first business message
+			// this will be used to notify users that they are talking with a human.
+			if ( firstBusinessMessageIndex === -1 && message.role === 'business' ) {
+				firstBusinessMessageIndex = index + 1;
+			}
```
Is this necessary? Based on the Figma it looks like this is only showing when the transfer to Zendesk happens. So when the user pulls up the conversation again later it shouldn't be there. It may be better to add a flag to the transfer message [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/odie-client/src/query/use-create-zendesk-conversation.ts#L22-L24) like `transfer-to-support` then use that flag to display the "Chatting with support now" message. That way when the conversation is loaded from the history that piece isn't there.

Also, this seems to be putting the component BEFORE the first user message but only after I send a response. I think it should be directly after the final bot response.

<img width="413" alt="Screen Shot 2024-10-29 at 11 24 29" src="https://github.com/user-attachments/assets/9dc406a7-e15f-462f-97e3-69c27bf2c16e">

renancarvalho replied:
> So when the user pulls up the conversation again later it shouldn't be there.

I thought otherwise, but I believe you're right. You gave a great idea on how to implement it. Thank you!

arcangelini commented on packages/odie-client/src/components/message/messages-container.tsx:
```diff
@@ -9,15 +10,23 @@ import type { CurrentUser } from '../../types/';
 }
+const DislikeThumb = () => {
+	return (
+		<div className="chatbox-message__dislike-thumb">
+			<ThumbsDown />
+		</div>
```
They styles for this override the message container styles when we show it as the user response. Mainly just the border radius I think?

<img width="104" alt="Screen Shot 2024-10-29 at 11 23 52" src="https://github.com/user-attachments/assets/857d6458-ca05-4be4-8b6e-376279c362dd">

renancarvalho replied:
I missed the border radius for sure, added.

renancarvalho committed:
Remove full-screen when using experiment

renancarvalho committed:
Fix odie response messages style

renancarvalho committed:
Rename classes

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Refactor support message

renancarvalho committed:
Fix thumbs_down border

renancarvalho committed:
Remove unnecessary code

renancarvalho committed:
Improve bot messages style

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
remove dislike style

arcangelini reviewed (approved):
Send it! :rocket:

agrullon95 commented:
The redesign looks great! Good job!

A few things I noticed compared to the figma designs:

* I think the color of the chat bubbles, links, send message button, etc should match the user's site colors. So in my case the color should be orange instead of blueberry. Let's confirm with @lucasmendes-design.

<img width="710" alt="CleanShot 2024-10-29 at 19 00 05@2x" src="https://github.com/user-attachments/assets/627986e0-d7f0-4f61-8291-1eb82879f0dc">


* The max width of the chat message (the content not including the avatar) should be 256px.
<img width="351" alt="CleanShot 2024-10-29 at 18 36 32@2x" src="https://github.com/user-attachments/assets/05e00031-7c40-412e-bb15-d310ba3be0f5">

* The thumbs-up/thumbs-down feedback icons should be smaller (probably just less padding)
<img width="352" alt="CleanShot 2024-10-29 at 18 38 17@2x" src="https://github.com/user-attachments/assets/0c36bc05-1286-4e40-976f-c6b6643f411c">

* The `chatting with support now` message should be below the last Wapuu message:
<img width="344" alt="CleanShot 2024-10-29 at 18 40 02@2x" src="https://github.com/user-attachments/assets/c55f4efb-219f-4981-a5e1-85289ab239d5">

agrullon95 commented on packages/odie-client/src/components/message/dislike-feedback-message.tsx:
```diff
@@ -9,32 +10,64 @@ import { uriTransformer } from './uri-transformer';
+						) }
+					</Markdown>
+				</div>
 				{ shouldUseHelpCenterExperience ? <GetSupport /> : extraContactOptions }
```
We don't need this condition since this only renders when `shouldUseHelpCenterExperience` is true.

agrullon95 commented on packages/odie-client/src/components/message/dislike-feedback-message.tsx:
```diff
@@ -9,32 +10,64 @@ import { uriTransformer } from './uri-transformer';
+							__i18n_text_domain__
+						) }
+					</Markdown>
+					{ shouldUseHelpCenterExperience ? <GetSupport /> : extraContactOptions }
```
Same as above. We don't need this condition anymore.

agrullon95 commented on packages/odie-client/src/components/message/get-support.tsx:
```diff
@@ -1,19 +1,23 @@
 	};
 
+	const getButtonText = () => {
+		return shouldUseHelpCenterExperience ? 'Get instant support' : 'Get support';
```
Let's translate this text.

renancarvalho replied:
Thank you. addressed

renancarvalho committed:
Merge branch 'trunk' into feat/redesign_helpcenter

renancarvalho committed:
Add translation

renancarvalho committed:
Remove unnecessary check

renancarvalho committed:
Add message max-width

renancarvalho committed:
Fix feedback button size

renancarvalho committed:
Fix spacing issues

renancarvalho committed:
Move support message label down

renancarvalho commented:
@agrullon95 thank you for reviewing, I believe I have addressed all your comments but this.

> We're missing the estimated time next to the Contact Support option, but it doesn't need to be added in this PR

Indeed, we are addressing it in a followup PR.

🙇

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16952507

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @renancarvalho, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Would you like to contact our support team? Select an option below:`
- `Get instant support`
- `Powered by Support AI. Some responses may be inaccurate`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16952507).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16952507).
